### PR TITLE
Upgrade the AwaDB from 0.3.5 to 0.3.6

### PR DESCRIPTION
**Pull request by ljeagle**
Upgrade the AwaDB from 0.3.5 to 0.3.6
1. Support get, delete and update batch documents
2. Add the implementation of the max_marginal_relevance_search
3. Change the default vectors' distance computation type from L2 to InnerProduct 
Please review @rlancemartin, @eyurtsev @hwchase17.  Thanks!

### CODE REVIEW

File: langchain/vectorstores/awadb.py [hunk 1]
```diff
@@ -110,7 +133,17 @@ def similarity_search(
         k: int = DEFAULT_TOPN,
         **kwargs: Any,
     ) -> List[Document]:
-        """Return docs most similar to query."""
+        """The most k similar documents of the specified text query.
```
Comment (efriis) [inline langchain/vectorstores/awadb.py:136]:
```suggestion
        """Returns the k most similar documents to the specified text query.
```
nit

File: langchain/vectorstores/awadb.py [hunk 2]
```diff
@@ -110,7 +133,17 @@ def similarity_search(
...
+
+        Args:
+            query: Text query
+            k: The most k similar documents of the text query
```
Comment (efriis) [inline langchain/vectorstores/awadb.py:140]:
```suggestion
            k: The maximum number of documents to return
```

File: langchain/vectorstores/awadb.py [hunk 3]
```diff
@@ -110,7 +133,17 @@ def similarity_search(
...
+            kwargs: Any possible extend parameters in the future.
+
+        Returns:
+            The most k similar documents of the specified text query
```
Comment (efriis) [inline langchain/vectorstores/awadb.py:144]:
```suggestion
            The k most similar documents to the specified text query
```

File: langchain/vectorstores/awadb.py [hunk 4]
```diff
@@ -172,9 +216,14 @@ def similarity_search_with_relevance_scores(
...
-        """Return docs and relevance scores, normalized on a scale from 0 to 1.
+        """Return docs the most similar to the text query.
+
+        Args:
+            query: Text query.
+            k: Number of the most similar documents to return. Defaults to 4.
 
-        0 is dissimilar, 1 is most similar.
+        Returns:
+            List of Documents the most similar to the text query.
```
Comment (efriis) [inline langchain/vectorstores/awadb.py:226]:
```suggestion
            List of (Document, relevance_score) tuples the most similar to the text query.
```

File: langchain/vectorstores/awadb.py [hunk 5]
```diff
@@ -172,9 +216,14 @@ def similarity_search_with_relevance_scores(
         k: int = DEFAULT_TOPN,
         **kwargs: Any,
     ) -> List[Tuple[Document, float]]:
-        """Return docs and relevance scores, normalized on a scale from 0 to 1.
+        """Return docs the most similar to the text query.
```
Comment (efriis) [inline langchain/vectorstores/awadb.py:219]: Was removing the portion about relevance scores intentional?

**Review by efriis — CHANGES_REQUESTED** [review]
Great changes! Main requested change is to align docstring on `similarity_search_with_relevance_scores` with the referenced types.

Looks like you removed the reference to relevance scores even though they're still present.

File: langchain/vectorstores/awadb.py (hunk 5 above)
Comment (ljeagle) [reply langchain/vectorstores/awadb.py:219]: Yes, the score has changed internally from L2 to InnerProduct, so the normalization can
not be needed again.

Comment (efriis) [reply langchain/vectorstores/awadb.py:219]: Could you add a note about what the float it's returning now with each document represents?

Comment (ljeagle) [reply langchain/vectorstores/awadb.py:219]: Good suggestion, I add a note about the changed relevance score.

**Comment by ljeagle** [comment]
@efriis I have added and changed the notes about relevance scores. please check, thanks!

**Comment by efriis** [comment]
@ljeagle looks like the formatter/lint check failed - could you update that and I can merge it in?

**Comment by ljeagle** [comment]
> @ljeagle looks like the formatter/lint check failed - could you update that and I can merge it in?

ok, thanks

**Review by efriis — APPROVED** [review]

**Comment by efriis** [comment]
Thanks @ljeagle !

**Pull request merged**